IA-2049 fix enketo edit on some sub sms additions

mestachs opened the pull request:
super seed https://github.com/BLSQ/iaso/pull/502
with the json parsing when turning xml to json 

IA-2049

beygorghor reviewed (approved):
This is working for me

olethanh commented on docker/webpack/Dockerfile:
```diff
@@ -11,9 +11,12 @@ RUN apt-get update && apt-get install -y \
 
 COPY package.json .
 COPY package-lock.json .
+COPY tsconfig.json .
+COPY babel-register.js .
+# COPY .babelrc .
```
These files should be  part of the files we mount in the docker-compose so change are reflected directly without needing to rebuild the image

olethanh commented:
I'm hitting this error in the eslintrc.json

```
Error: .eslintrc.json#overrides[0]:
	Configuration for rule "import/no-unresolved" is invalid:
	Value {"ignores":["modules"],"caseSensitive":true} should NOT have additional properties.


Error: .eslintrc.json#overrides[0]:
	Configuration for rule "import/no-unresolved" is invalid:
	Value {"ignores":["modules"],"caseSensitive":true} should NOT have additional properties.

    at ConfigValidator.validateRuleOptions (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:144:23)
    at /home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:199:18
    at Array.forEach (<anonymous>)
    at ConfigValidator.validateRules (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:196:34)
    at ConfigValidator.validateConfigArray (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:322:18)
    at CascadingConfigArrayFactory._finalizeConfigArray (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:493:23)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/home/olivier/Projects/blsq/iaso/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:299:21)
    at CLIEngine.isPathIgnored (/home/olivier/Projects/blsq/iaso/node_modules/eslint/lib/cli-engine/cli-engine.js:973:18)
    at ESLintPlugin.invokeESLint (/snap/pycharm-professional/265/plugins/JavaScriptLanguage/languageService/eslint/bin/eslint-plugin.js:89:23)
    at ESLintPlugin.getErrors (/snap/pycharm-professional/265/plugins/JavaScriptLanguage/languageService/eslint/bin/eslint-plugin.js:67:21)
Process finished with exit code -1
```

olethanh commented:
I tried porting date.js file to Typescript. It works well. 
But I get a warning because it would like the extensions in the import `ESLint: Missing file extension "ts" for "../../utils/dates"(import/extensions)` not sure this is what we want? if it's not maybe we can deactivate this rule?

![image](https://user-images.githubusercontent.com/82500/146161568-79a3038e-6cc9-445a-978f-6a13fe6b38bf.png)

olethanh committed:
IA-2049 fix Bug with special char in enketo edit

The problem was that BeautifulSoup didn't parse the xml properly
and so couldn't get the instanceId.

Converted the code to only parse once using xml.etree instead of three
time

Test don't pass because it's missing namespace now

olethanh committed:
IA-2049 Revert to lxml so the namespace is keep

xml.etree was droping the namespace attribute from the root
tag when reconverting back to xml, so revert to using lxml
that does it properly without fuss

olethanh committed:
IA-2049 Add test to check for regression

olethanh committed:
IA-2049 Correct previous test so they still check something

olethanh committed:
IA-2049 better function signature

olethanh committed:
IA-2049 fix mistake

olethanh committed:
Fix typing

olethanh committed:
mypy

mestachs committed:
Make sure gsoup is not playing with the encoding

mestachs committed:
Try to make mypy happy

mestachs committed:
Black happy

mestachs committed:
Add test with special char

madewulf reviewed (approved):
LGTM
